### PR TITLE
#571 Use pre-commit dog food and update release process for pre-commit autoupdate.

### LINKED ISSUE #571: pre-commit autoupdate seems to fail
When I launch pre-commit autoupdate I get this

```
$ pre-commit autoupdate
[https://github.com/pre-commit/pre-commit-hooks] already up to date!
[https://github.com/psf/black-pre-commit-mirror] already up to date!
[https://github.com/adamchainz/blacken-docs] already up to date!
[https://github.com/rstcheck/rstcheck] already up to date!
[https://github.com/charliermarsh/ruff-pre-commit] already up to date!
[https://github.com/nbQA-dev/nbQA] already up to date!
[https://github.com/twisted/towncrier]
=====> /var/folders/2z/142033n17rbfy969s6h4hymw0000gn/T/tmphs0r_674/.pre-commit-hooks.yaml is not a file
```

So all other git hooks get updated, but it crashes only when trying to update this project.

I also tried to manually update the config file with what I see from your docs,

```yaml
  - repo: https://github.com/twisted/towncrier
    rev: 22.13.0
    hooks:
      - id: towncrier-check
```

but the result doesn't change.

Also `pre-commit clean` doesn't seem to help.

### PR DESCRIPTION
**Pull request by adiroiban**
# Description
Fixes: #571

Try to reproduce the reported error.

Besides using `pre-commit` as part of our CI

this updates the release process to make it work with `pre-commit auto-update`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -194,6 +194,28 @@ jobs:
...
+      with:
+        fetch-depth: 0
+
+    - name: Set up python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.12
+
+    - name: Install dependencies
+      run: python -m pip pre-commit
```
Comment (hynek) [inline .github/workflows/ci.yml]:
```suggestion
      run: python -m pip install pre-commit
```

**Comment by hynek** [comment]
I don't think https://github.com/twisted/towncrier/issues/571 is on our side tbh, but dogfooding is good regardless.

**Comment by adiroiban** [comment]
I don't have time to look into this.

If anyone cares about pre-commit, feel free to send a fix.

This PR demonstrates the issue and I hope that it can be use to test the fix :)

**Comment by adiroiban** [comment]
This is ready for review.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -195,6 +195,36 @@ jobs:
       if: ${{ matrix.task.run-if }}
 
 
+  pre-commit:
+    name: Check pre-commit integration
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (SmileyChris) [inline .github/workflows/ci.yml]:
```suggestion
    - uses: actions/checkout@v4
```
This should be done across the board really since v3, but that's can be saved for a separate PR :)

**Review by SmileyChris — APPROVED** [review]
The pre-commit.rst docs fix is especially important, since I think the previous docs were actually broken with the rev they were on...
LGTM

**Comment by adiroiban** [comment]
Thanks for the review and feedback. I have triggered auto-merge on this PR.

I have reconfigured the repo to allow merge commits

image

-------

Previously, only squash merges were allowed
